File: snippets/fsharp/System/Convert/ChangeType/convertchangetype.fs
module convertchangetype

//<snippet1>
open System

let d = -2.345
let i = Convert.ChangeType(d, typeof<int>) :?> int

printfn $"The double value {d} when converted to an int becomes {i}"

let s = "12/12/98"
let dt = Convert.ChangeType(s, typeof<DateTime>) :?> DateTime

printfn $"The string value {s} when converted to a Date becomes {dt}"
//</snippet1>

File: snippets/csharp/System/Convert/ChangeType/convertchangetype.cs
//<snippet1>
using System;

public class ChangeTypeTest
{
    public static void Main()
    {

        double d = -2.345;
        int i = (int)Convert.ChangeType(d, typeof(int));

        Console.WriteLine($"The double value {d} when converted to an int becomes {i}");

        string s = "12/12/98";
        DateTime dt = (DateTime)Convert.ChangeType(s, typeof(DateTime));

        Console.WriteLine($"The string value {s} when converted to a Date becomes {dt}");
    }
}
//</snippet1>
